mesheryctl/internal/cli/root/system/model.go

```diff
@@ -246,24 +246,119 @@ mesheryctl system model view [model-name]
 	},
 }
 
-// ModelCmd represents the `mesheryctl system model` command
+// represents the `mesheryctl exp model search [query-text]` subcommand.
+var searchModelCmd = &cobra.Command{
+	Use:   "search",
+	Short: "search models",
+	Long:  "search a models by search string",
+	Example: `
+// View current provider
+mesheryctl exp model search [query-text]
+	`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		//Check prerequisite
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return err
+		}
+		err = utils.IsServerRunning(mctlCfg.GetBaseMesheryURL())
+		if err != nil {
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return err
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+	Args: func(_ *cobra.Command, args []string) error {
+		const errMsg = "Usage: mesheryctl exp model search [query-text]\nRun 'mesheryctl exp model search --help' to see detailed help message"
+		if len(args) == 0 {
+			return fmt.Errorf("query text must not be empty\n\n%v", errMsg)
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			log.Fatalln(err, "error processing config")
+		}
+
+		baseUrl := mctlCfg.GetBaseMesheryURL()
+		queryText := args[0]
+
+		url := fmt.Sprintf("%s/api/meshmodels/models?search=%s&pagesize=all", baseUrl, queryText)
+		req, err := utils.NewRequest(http.MethodGet, url, nil)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		resp, err := utils.MakeRequest(req)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		// defers the closing of the response body after its use, ensuring that the resources are properly released.
+		defer resp.Body.Close()
+
+		data, err := io.ReadAll(resp.Body)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		modelsResponse := &models.MeshmodelsAPIResponse{}
+		err = json.Unmarshal(data, modelsResponse)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		header := []string{"Category", "Model", "Version"}
+		rows := [][]string{}
+
+		for _, model := range modelsResponse.Models {
+			if len(model.DisplayName) > 0 {
+				rows = append(rows, []string{model.Category.Name, model.Name, model.Version})
+			}
+		}
+
+		if len(rows) == 0 {
+			// if no model is found
+			utils.Log.Info("No model(s) found")
```

Comment:
What about components? Are we not offering search support for components, too?

Reply:
@leecalcote  I am calling this API [endpoint](https://github.com/meshery/meshery/blob/0dd426525ab7dda847dd02b5c0b1bbf4721c6127/server/router/server.go#L187) to search models and this endpoint utilizes [GetModels](https://github.com/meshery/meshkit/blob/0133432a3ce871fdc12b148745788d3813a5afe5/models/meshmodel/registry/registry.go#L290) meshkit function. this function doesn't search for components (what I learnt by reading codebase).

Reply:
if we have to increase capibility of search command to search for components as well then I will provide a new flag for components e.g. `--components [query-text] --components [query-text]`, sounds good?